ex1/buttonActions.js

```diff
@@ -1,13 +1,50 @@
 
-var indexOfColor = 1;
-function changeColor() {
-  var colorArray = ['#FF6633', '#CC9999','#4D80CC'];
-    //store
-    sessionStorage.setItem("color", colorArray[indexOfColor]);
-    // Retrieve
-    document.getElementById("demo").style.backgroundColor = sessionStorage.getItem("color");
-
-    if(++indexOfColor == colorArray.length){
-      indexOfColor = 0;
-    }
+const INITIAL_COLOR = 0;
+const colorArray = ['blue', 'red','orange','yellow'];
+let indexOfColor = INITIAL_COLOR;
+let weGotToEndOfTheArrayFlag = false;
+
+function handleButtonAndImageStyles(){
+   handleButtonColor();
+   handleImageDisplay();
+}
+
+function handleButtonColor(){
+  setButtonColor();
```

Comment:
no way in the world you first set a color and then update the index.  Please set the color at the end of playing around with the index